test/product-import.spec.coffee

```diff
@@ -510,6 +510,19 @@ describe 'ProductImport unit tests', ->
         done()
       .catch done
 
+    it 'should throw error in case of too big query', (done) ->
+      exptectedErrorMessage = 'product fetch query size: 10952 bytes, exceeded the supported size, please try with a smaller batch size.'
```

Comment:
For A could be a problem if product to update has  master variant as completely new variant with new sku and old master does not exist as variant anymore.

For B see problem A